### ISSUE TITLE
It seems that Eq 7-8 are both performed twice in the rendering pipeline

### ISSUE THREAD

**Issue by kaname-madoka18** [oppo-us-research/SpacetimeGaussians#43]
I wonder if line 78-91 in file .\thirdparty\gaussian_splatting\renderer\__init__.py and line 192-200 in file .\thirdparty\gaussian_splatting\submodules\forward_full\cuda_rasterizer\forward.cu do exactly the same thing, but twice
![image](https://github.com/oppo-us-research/SpacetimeGaussians/assets/43375270/ddba9bd6-884c-4233-b463-cc5685affa23)
![image](https://github.com/oppo-us-research/SpacetimeGaussians/assets/43375270/a95007f3-e581-488d-8527-ba83fda98b71)

**Comment by lizhan17** [comment]
sorry, there is some confusions.  ```train_ours_full``` is for training our full model. 

while in ```forward_full``` cuda code, it is for testing our full model, which fuses the python code into cuda for faster rendering. 
this cuda code should work with ```test_ours_fullss_fused``` 

you can simply modify the ```test_ours_fullss_fused``` function to get ```test_ours_full_fused``` by removing the grid sampling parts.  i will upload the function ```test_ours_full_fused``` in the next few days.

**Comment by lizhan17** [comment]
this should be ```test_ours_full_fused ``` function.

it seems to make a bit more modification of repo for sigmoid function option for n3d and technicolor. will try to handle it if i have time.  or just another ```test_ours_full_fused_sigmoid```. 
 
```
def test_ours_full_fused(viewpoint_camera, pc : GaussianModel, pipe, bg_color : torch.Tensor, scaling_modifier = 1.0, override_color = None, basicfunction = None, GRsetting=None, GRzer=None):

 
    # Create zero tensor. We will use it to make pytorch return gradients of the 2D (screen-space) means
    screenspace_points = torch.zeros_like(pc.get_xyz, dtype=pc.get_xyz.dtype, requires_grad=True, device="cuda") + 0


    torch.cuda.synchronize()
    startime = time.time()#perf_counter

    tanfovx = math.tan(viewpoint_camera.FoVx * 0.5)
    tanfovy = math.tan(viewpoint_camera.FoVy * 0.5)

    raster_settings = GRsetting(
        image_height=int(viewpoint_camera.image_height),
        image_width=int(viewpoint_camera.image_width),
        tanfovx=tanfovx,
        tanfovy=tanfovy,
        bg=bg_color,
        scale_modifier=scaling_modifier,
        viewmatrix=viewpoint_camera.world_view_transform,
        projmatrix=viewpoint_camera.full_proj_transform,
        sh_degree=pc.active_sh_degree,
        campos=viewpoint_camera.camera_center,
        prefiltered=False,
        debug=False)

    rasterizer = GRzer(raster_settings=raster_settings)
    
    


    tforpoly = viewpoint_camera.timestamp - pc.get_trbfcenter
    rotations = pc.get_rotation(tforpoly) # to try use 
    colors_precomp = pc.get_features(tforpoly)

    
    means2D = screenspace_points
   

    cov3D_precomp = None

    #scales = pc.get_scaling

    shs = None
    # sandwichold 2,3
    rendered_image, radii = rasterizer(
        timestamp = viewpoint_camera.timestamp, 
        trbfcenter = pc.get_trbfcenter,
        trbfscale = pc.computedtrbfscale ,
        motion = pc._motion,
        means3D = pc.get_xyz,
        means2D = means2D,
        shs = shs,
        colors_precomp = colors_precomp,
        opacities = pc.computedopacity,
        scales = pc.computedscales,
        rotations = rotations,
        cov3D_precomp = cov3D_precomp,
        mlp1 = pc.rgbdecoder.mlp1.weight.squeeze(3).squeeze(2).flatten(),
        mlp2 = pc.rgbdecoder.mlp2.weight.squeeze(3).squeeze(2).flatten(),
        rayimage = viewpoint_camera.rays)
    # add sigmoid for n3d dataset  
    torch.cuda.synchronize()
    duration = time.time() - startime #

    # INTER_CUBIC
    rendered_image = rendered_image.squeeze(0)


    return {"render": rendered_image,
            "viewspace_points": screenspace_points,
            "visibility_filter" : radii > 0,
            "radii": radii,
            "rawrendered_image": rendered_image,
             "duration": duration}


```

**Comment by lizhan17** [comment]
added ```test_ours_full_fused``` pipeline which fuse the mlp into the cuda rasterizer for the nerual3d and technicolor dataset. 
please reopen if you still have problem. i wanted to reduce the number of opened issue before the cvpr...
